services/order.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, User
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime = None
+) -> Order:
+    with transaction.atomic():
+        order = Order()
+        user = User.objects.get(username=username)
+        order.user = user
+        if date:
+            order.save()
+            order.created_at = date
+        order.save()
```

Comment:
The logic for setting the `created_at` date is incorrect. The `created_at` field should be set before the first `order.save()` call. Consider setting `order.created_at = date` before saving the order for the first time.